lusidtools/cocoon/utilities.py

```diff
@@ -1121,5 +1122,155 @@ def create_scope_id(time_generator=None):
     # Multiply by 7 to get value to 100s of nano seconds
     timestamp = hex(int(test * 10000000.0))
     # Create the scope id by joining the hex representation with dashes every 4 characters
-    scope_id = "-".join(timestamp[i : i + 4] for i in range(2, len(timestamp), 4))
+    scope_id = "-".join(timestamp[i: i + 4] for i in range(2, len(timestamp), 4))
     return scope_id
+
+
+def default_fx_forward_model(df: pd.DataFrame, FX_code: str, func_p, func_l,
+                             mapping: dict) -> (pd.DataFrame, dict):
+    """
+    Function that takes 2 rows representing a single forward and merge them into a single transaction
+
+    :param df:  DataFrame containing transactions data
+    :param T_ID:  Column name of transactions IDs
+    :param transaction_type:  Column name of transaction types
+    :param FX_code:  The transaction type that identifies a forward
+    :param func_p:  function that evaluates to true for where the dataframe row is a buy (incoming) side
```

Comment:
I think you can annotate this with a `Callable` type

Reply:
changed to typing.Callable[[], bool]